aptos-move/framework/aptos-framework/sources/block.move

```diff
@@ -178,6 +180,31 @@ module aptos_framework::block {
         );
     }
 
+    ///  Emit a `NewBlockEvent` event. This function will be invoked by write set script directly to generate the
+    ///  new block event for WriteSetPayload.
+    public fun emit_writeset_block_event(framework_signer: &signer) acquires BlockResource {
+        system_addresses::assert_aptos_framework(framework_signer);
+
+        let block_metadata_ref = borrow_global_mut<BlockResource>(@aptos_framework);
+        block_metadata_ref.height = event::counter(&block_metadata_ref.new_block_events);
+
+        let genesis_id = @0x0;
+        emit_new_block_event(
+            framework_signer,
+            &mut block_metadata_ref.new_block_events,
+            NewBlockEvent {
+                hash: genesis_id,
+                epoch: reconfiguration::current_epoch(),
+                round: MAX_U64,
+                height: block_metadata_ref.height,
+                previous_block_votes_bitvec: vector::empty(),
+                proposer: @vm_reserved,
+                failed_proposer_indices: vector::empty(),
```

Comment:
Would this become a problem as this would wipe all previous failed proposer indices? @zekun000

Reply:
Should we just use the old value here?

Reply:
this is only the value for this specific fake block